run.py

```diff
@@ -0,0 +1,153 @@
+import argparse
+import csv
+from pathlib import Path
+import os
+import pickle
+import yaml
+
+import pandas as pd
+from sklearn.feature_extraction.text import CountVectorizer
+from bertopic.vectorizers import ClassTfidfTransformer
+from spacy.lang.en import stop_words
+from sentence_transformers import SentenceTransformer
+
+from PatientX.models.BERTopicModel import BERTopicModel
+from PatientX.MistralRepresentation import MistralRepresentation
+
+
+def read_data(datafolder: Path):
+    """
+    Read in data
+
+    datafolder: Path
+    Returns: List containing documents
+    """
+    dfs = []
+
+    assert(os.path.isdir(datafolder))
+
+    for filename in os.listdir(datafolder):
+        filepath = os.path.join(datafolder, filename)
+        if os.path.isfile(filepath):
+            try:
+                df = pd.read_csv(filepath)
+                dfs.append(df.copy(deep=True))
+            except csv.Error as e:
+                pass
+
+    full_dataset = pd.concat(dfs, ignore_index=True)
+    cleaned_text = pd.DataFrame()
+    try:
+        grouped_dataset = full_dataset.groupby(['forum', 'thread_title', 'message_nr'], as_index=False).agg(
+            {'post_message': ''.join})
+
+        grouped_dataset['post_message'] = grouped_dataset['post_message'].str.strip().replace(r'\n', ' ', regex=True)
+        cleaned_text = grouped_dataset['post_message'].replace({r'\s+$': '', r'^\s+': ''}, regex=True).replace(r'\n', ' ',
+                                                                                                           regex=True)
+        cleaned_text = cleaned_text.replace(r'\t', ' ', regex=True)
+        cleaned_text = cleaned_text.replace(r'\r', ' ', regex=True)
+    except KeyError as e:
+        raise KeyError("Check README file for proper data format")
+
+    return cleaned_text.tolist()
+
+
+def get_count_vectorizer():
+    custom_stop_words = list(stop_words.STOP_WORDS)
+    custom_stop_words += ['with', 'my', 'your', 'she', 'this', 'was', 'her', 'have', 'as', 'he', 'him', 'but', 'not',
+                          'so', 'are', 'at', 'be', 'has', 'do', 'got', 'how', 'on', 'or', 'would', 'will', 'what',
+                          'they', 'if', 'or', 'get', 'can', 'we', 'me', 'can', 'has', 'his', 'there', 'them', 'just',
+                          'am', 'by', 'that', 'from', 'it', 'is', 'in', 'you', 'also', 'very', 'had', 'a', 'an', 'for']
+
+    vectorizer_model = CountVectorizer(stop_words=custom_stop_words)
+
+    return vectorizer_model
+
+
+def get_ctidf_model():
+    return ClassTfidfTransformer(reduce_frequent_words=True)
+
+
+def get_medical_embedding_model():
+    return SentenceTransformer('pritamdeka/S-PubMedBert-MS-MARCO')
+
+
+def get_representation_model(nr_docs=10, document_diversity=0.1):
+    representation_model = MistralRepresentation(nr_docs=nr_docs, diversity=document_diversity, api="generate")
+
+    return representation_model
+
+
+def main():
+    if not os.path.isfile("config.yaml"):
```

Comment:
Requiring a yaml file feels wonky to me. It would be better to take command line arguments rather than requiring a yaml file. These arguments can have defaults to that they aren't all required. If you use the `typer` package, you can also use the `typer-config` package to invoke the script with a yaml file if they want to (which would be the best of both worlds).